packages/compass-databases-navigation/src/databases-navigation-tree.tsx

```diff
@@ -0,0 +1,744 @@
+  '0%': {
+    opacity: 0,
+  },
+  '100%': {
+    opacity: 1,
+  },
+});
+
+const fadeIn = css({
+  animation: `${fadeInAnimation} .16s ease-out`,
+});
+
+const collectionList = css({
+  contentVisibility: 'auto',
+});
+
+const CollectionsList: React.FunctionComponent<{
+  collections: Collection[];
+  onNamespaceAction(namespace: string, action: Actions): void;
+  activeNamespace?: string;
+  isReadOnly: boolean;
+}> = ({ collections, onNamespaceAction, activeNamespace, isReadOnly }) => {
+  return (
+    <ul
+      role="group"
+      className={cx(collectionList, ulReset)}
+      style={{
+        // This exists, but react types are not aware
+        // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+        // @ts-expect-error
+        containIntrinsicSize: COLLECTION_ROW_HEIGHT * collections.length,
+      }}
+    >
+      {collections.map((coll, index) => (
+        <CollectionItem
+          key={coll._id}
+          id={coll._id}
+          name={coll.name}
+          type={coll.type}
+          posInSet={index + 1}
+          setSize={collections.length}
+          onNamespaceAction={onNamespaceAction}
+          isActive={coll._id === activeNamespace}
+          isReadOnly={isReadOnly}
+        />
+      ))}
+    </ul>
+  );
+};
+
+const placeholderContainer = css({
+  position: 'absolute',
+  pointerEvents: 'none',
+  top: 0,
+  right: 0,
+  bottom: 0,
+  left: 0,
+  opacity: 1,
+});
+
+const placeholderContainerHidden = css({
+  opacity: 0,
+  transition: 'opacity .16s ease-out',
+});
+
+const placeholderItem = css({
+  display: 'flex',
+  alignItems: 'center',
+  height: COLLECTION_ROW_HEIGHT,
+  paddingLeft: spacing[5],
+});
+
+const placeholderItemContent = css({
+  display: 'block',
+  height: spacing[3],
+  backgroundColor: uiColors.gray.dark1,
+  borderRadius: 3,
+});
+
+function getBoundRandom(min: number, max: number) {
+  return Math.random() * (max - min) + min;
+}
+
+const PlaceholderItem: React.FunctionComponent = () => {
+  const width = useMemo(() => {
+    return `${getBoundRandom(30, 80)}%`;
+  }, []);
+
+  return (
+    <li className={placeholderItem}>
+      <span style={{ width }} className={placeholderItemContent}></span>
+    </li>
+  );
+};
+
+const CollectionsListPlaceholder: React.FunctionComponent<{
+  length: number;
+  isVisible: boolean;
+}> = ({ length, isVisible }) => {
+  const items = useMemo(() => {
+    return Array.from({ length }, (_, index) => (
+      <PlaceholderItem key={index}></PlaceholderItem>
+    ));
+  }, [length]);
+
+  return (
+    <ul
+      role="presentation"
+      data-testid="placeholder"
+      className={cx(
+        ulReset,
+        placeholderContainer,
+        !isVisible && placeholderContainerHidden
+      )}
+    >
+      {items}
+    </ul>
+  );
+};
+
+const actionsContainer = css({
+  flex: 'none',
+  marginLeft: 'auto',
+  alignItems: 'center',
+  display: 'flex',
+});
+
+const databaseItem = css({
+  height: DATABASE_ROW_HEIGHT,
```

Comment:
Currently in Compass the database and collections tree items have a `cursor`: `pointer` on hover, do we want to keep that functionality? I have no preference.